tests/integration-tests/tests/api_infrastructure/test_api_infrastructure.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import os
+import re
+
+import boto3
+import pytest
+from assertpy import assert_that, soft_assertions
+from cfn_stacks_factory import CfnStack, CfnStacksFactory
+from utils import generate_stack_name
+
+
+@pytest.fixture()
+@pytest.mark.usefixtures("setup_sts_credentials")
+def api_infrastructure_stack_factory(request):
```

Comment:
shall we move this fixture to conftest and have it in common with the networking test? In the end what we need is a cfn_stacks_factory that has a class scope rather than session (that already exists in conftest). we could rename the other one to session_cfn_stacks_factory or find a meaningful name for the new one.

Reply:
Agreed, I was thinking of moving it to utils, but conftest is better since it's where I have put the other two fixtures